docs/source/upgrading_your_components.md

```diff
@@ -0,0 +1,262 @@
+
+In addition to the YAML configuration file of the binary, you will also have to set the environment variables for your deployment. More on this in the next section.
+
+## Environment variables for the binaries
+
+When you deploy a peer or an ordering node, you had to set a number of environment variables relevant to its configuration. A best practice is to create a file for these environment variables, give it a name relevant to the node being deployed, and save it somewhere on your local file system. That way you can be sure that when upgrading the peer or ordering node you are using the same variables you set when creating it by issuing:
+
+```
+docker run --env-file ./env<name of node>.list ubuntu bash
+```
+
+Here's a list of some of the **peer** environment variables (with sample values --- as you can see from the addresses, these environment variables are for a network deployed locally) that can be set that be listed in the `./env<name of node>.list` file:
+
+```
+CORE_PEER_TLS_ENABLED=true
+CORE_PEER_GOSSIP_USELEADERELECTION=true
+CORE_PEER_GOSSIP_ORGLEADER=false
+CORE_PEER_PROFILE_ENABLED=true
+CORE_PEER_TLS_CERT_FILE=/etc/hyperledger/fabric/tls/server.crt
+CORE_PEER_TLS_KEY_FILE=/etc/hyperledger/fabric/tls/server.key
+CORE_PEER_TLS_ROOTCERT_FILE=/etc/hyperledger/fabric/tls/ca.crt
+CORE_PEER_ID=peer0.org1.example.com
+CORE_PEER_ADDRESS=peer0.org1.example.com:7051
+CORE_PEER_LISTENADDRESS=0.0.0.0:7051
+CORE_PEER_CHAINCODEADDRESS=peer0.org1.example.com:7052
+CORE_PEER_CHAINCODELISTENADDRESS=0.0.0.0:7052
+CORE_PEER_GOSSIP_BOOTSTRAP=peer0.org1.example.com:7051
+CORE_PEER_GOSSIP_EXTERNALENDPOINT=peer0.org1.example.com:7051
+CORE_PEER_LOCALMSPID=Org1MSP
+```
+
+Here are some **ordering node** variables (again, these are sample values) that might be listed in the environment variable file for a node:
+
+```
+ORDERER_GENERAL_LISTENADDRESS=0.0.0.0
+ORDERER_GENERAL_GENESISMETHOD=file
+ORDERER_GENERAL_GENESISFILE=/var/hyperledger/orderer/orderer.genesis.block
+ORDERER_GENERAL_LOCALMSPID=OrdererMSP
+ORDERER_GENERAL_LOCALMSPDIR=/var/hyperledger/orderer/msp
+ORDERER_GENERAL_TLS_ENABLED=true
+ORDERER_GENERAL_TLS_PRIVATEKEY=/var/hyperledger/orderer/tls/server.key
+ORDERER_GENERAL_TLS_CERTIFICATE=/var/hyperledger/orderer/tls/server.crt
+ORDERER_GENERAL_TLS_ROOTCAS=[/var/hyperledger/orderer/tls/ca.crt]
+ORDERER_GENERAL_CLUSTER_CLIENTCERTIFICATE=/var/hyperledger/orderer/tls/server.crt
+ORDERER_GENERAL_CLUSTER_CLIENTPRIVATEKEY=/var/hyperledger/orderer/tls/server.key
+ORDERER_GENERAL_CLUSTER_ROOTCAS=[/var/hyperledger/orderer/tls/ca.crt]
```

Comment:
The full list of typical env variables would make more sense in a 'running a peer from docker images' topic, not the upgrade topic. But I don't think we have anything like that yet...